pyscraper/new_hansard.py

```diff
@@ -0,0 +1,1442 @@
+        if self.current_speech is not None:
+            self.clear_current_speech()
+
+        tag = etree.Element('division')
+
+        tag.set('id', self.get_speech_id())
+        tag.set('nospeaker', 'true')
+        tag.set('divdate', self.date)
+        div_number = \
+            division.xpath('.//ns:Number/text()', namespaces=self.ns_map)
+
+        tag.set('divnumber', u''.join(div_number))
+        tag.set('colnum', self.current_col)
+        tag.set('time', self.current_time)
+
+        ayes_count = \
+            division.xpath('.//ns:AyesNumber/text()', namespaces=self.ns_map)
+        noes_count = \
+            division.xpath('.//ns:NoesNumber/text()', namespaces=self.ns_map)
+
+        div_count = etree.Element('divisioncount')
+        div_count.set('ayes', u''.join(ayes_count))
+        div_count.set('noes', u''.join(noes_count))
+
+        tag.append(div_count)
+
+        ayes = division.xpath(
+            './/ns:NamesAyes//ns:Member', namespaces=self.ns_map
+        )
+        noes = division.xpath(
+            './/ns:NamesNoes//ns:Member', namespaces=self.ns_map
+        )
+
+        aye_tellers = division.xpath(
+            './/ns:TellerNamesAyes//ns:Member', namespaces=self.ns_map
+        )
+        noe_tellers = division.xpath(
+            './/ns:TellerNamesNoes//ns:Member', namespaces=self.ns_map
+        )
+
+        aye_list = etree.Element('mplist')
+        aye_list.set('vote', 'aye')
+        aye_list = self.parse_votelist(ayes, 'aye', aye_list)
+        aye_list = self.parse_votelist(aye_tellers, 'aye', aye_list, True)
+        tag.append(aye_list)
+
+        noe_list = etree.Element('mplist')
+        noe_list.set('vote', 'no')
+        noe_list = self.parse_votelist(noes, 'no', noe_list)
+        noe_list = self.parse_votelist(noe_tellers, 'no', noe_list, True)
+        tag.append(noe_list)
+
+        self.root.append(tag)
+        self.current_speech_num = self.current_speech_num + 1
+
+    def parse_time(self, tag):
+        time_txt = u''.join(tag.xpath('.//text()'))
+        if time_txt == '':
+            return
+        matches = re.match('(\d+)(?:\.(\d+))?\s*(am|pm)', time_txt)
+        if matches:
+            hours = int(matches.group(1))
+            minutes = int(matches.group(2) or 0)
+            # mmmmmm
+            if matches.group(3) == 'pm':
+                if hours != 12:
+                    hours = hours + 12
+            time = mx.DateTime.DateTimeFrom(hour=hours, minute=minutes)
+            self.current_time = time.strftime('%H:%M:%S')
+        else:
+            raise ContextException(
+                "Unmatched time %s" % time_txt,
+                fragment=tag,
+                stamp=tag.get('url')
+            )
+
+    def parse_pi(self, pi):
+        # you would think there is a better way to do this but I can't seem
+        # to extract attributes from processing instructions :(
+        text = str(pi)
+        matches = re.search(r'column=(\d+)\?', text)
```

Comment:
This doesn't appear to be catching all column changes, e.g. through a division, and speeches after the division have an incorrect column number (and less so, ID) until the next column during a speech. Does this function need to be some sort of preprocessor to all other handling?